src/network-layer/default/RelayDefaultNetworkLayer.js

```diff
@@ -89,7 +89,8 @@ class RelayDefaultNetworkLayer {
         }
       }).catch(
         error => request.reject(error)
-      )
+      ),
+      request
```

Comment:
Ok let me explain: Here is my code: I want to intercept query and mutation requests. 
```Javascript
Relay.injectNetworkLayer(
      {
        ...defaultNetworkLayer,
        sendQueries: (queryRequests) => {
          return defaultNetworkLayer.sendQueries(queryRequests).then(
            response => console.log(response),
            error => console.error(error, 'ERR'),
          )
        }
      }
    );
```
So i pass array of requests to defaultNetworkLayer.sendQueries and im gonna waiting for a promise.
So the sendQueries takes that array, takes each element in array (instance of RelayQueryRequest) resolves or rejects it and return a different promise.  So you don't resolve that promise which you return, you resolve RelayQueryRequest. I can make example in plnk or smth else.